shenyu-plugin/shenyu-plugin-motan/src/main/java/org/apache/shenyu/plugin/motan/proxy/MotanProxyService.java

```diff
@@ -46,6 +49,8 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(MotanProxyService.class);
 
+    private final ExecutorService threadPool = Executors.newCachedThreadPool(ShenyuThreadFactory.create("shenyu-motan", true));
```

Comment:
for your mail, when we have a large number of requests, cached thread pool may create Integer.MAX_VALUE size threads, may cause oom

Reply:
> 

you can see `org.apache.dubbo.common.threadpool.support.cached.CachedThreadPool`, dubbo also use a `CachedThreadPool`, so it shouldn't be a problem.

In the longer term, this thread pool will be replaced by a shared thread pool(you can see https://github.com/apache/incubator-shenyu/issues/3026).

Reply:
> for your mail, when we have a large number of requests, cached thread pool may create Integer.MAX_VALUE size threads, may cause oom

you can see `org.apache.dubbo.common.threadpool.support.cached.CachedThreadPool`, dubbo also use a `CachedThreadPool`, so it shouldn't be a problem.

In the longer term, this thread pool will be replaced by a shared thread pool(you can see https://github.com/apache/incubator-shenyu/issues/3026).

Reply:
how about use ShenyuThreadPoolExecutor here?

Reply:
> Dubbo's cached thread pool support config maxThreads and maxQueue, as it can used to protect dubbo process, so I suggest we config maxThreads and maxQueue like ShenyuThreadPoolExecutor to protect bootstrap process from oom.

`ShenyuThreadPoolExecutor` now supports configuring the maximum number of threads. You can see it on `shenyu-bootstrap/src/main/resources/application.yml`:
```yml
#  sharedPool:
#    enable: true
#    prefix: "shenyu-shared"
#    corePoolSize: 200
#    maximumPoolSize: 2000
#    keepAliveTime: 60000
#    # 1GB
#    maxWorkQueueMemory: 1073741824
```

Reply:
> Dubbo's cached thread pool support config maxThreads and maxQueue, as it can used to protect dubbo process, so I suggest we config maxThreads and maxQueue like ShenyuThreadPoolExecutor to protect bootstrap process from oom.

I planed to complete the transformation of the shared thread pool in the following order:
1. use CachedThreadPool as default thread pool(this PR);
2. support config thread pool type, core threads, max threads and max queues;
3. add support for `ShenyuThreadPoolExecutor`;

Because `2.4.3` will be released soon, only one part will be submitted for risk consideration.

Reply:
> Dubbo's cached thread pool support config maxThreads and maxQueue, as it can used to protect dubbo process, so I suggest we config maxThreads and maxQueue like ShenyuThreadPoolExecutor to protect bootstrap process from oom.

I planed to complete the transformation of the shared thread pool in the following order:
1. use `CachedThreadPool` as the default thread pool (this PR);
2. support config thread pool type, core threads, max threads and max queues;
3. add support for `ShenyuThreadPoolExecutor`;

Because `2.4.3` will be released soon, only one part will be submitted for risk consideration.

Reply:
Your plan sounds good. Can we use a compromise method which won't cause oom or reduce it and can also improve performance. How about new a CachedThreadPool with a limited maxPoolSize.

Reply:
> Your plan sounds good. Can we use a compromise method which won't cause oom or reduce it and can also improve performance. How about new a CachedThreadPool with a limited maxPoolSize.

Due to the different hardware configuration of the machine, it is impossible to give a constant value. I think it is a better choice to combine `part 1` and `part 2`, how did you feel ?

Reply:
Yes, I agree with u.